### PR TITLE
chore(CI): Refactor and re-enable simulator nightly

### LINKED ISSUE #4290: CI: re-enable `simulator_nightly.yml`

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Removes the teleport functionality from the simulator nightly workflow and refactors it so that the tests are run on our self hosted runner.

Removes the reporting job as this is already handled elsewhere.

## Links to any relevant issues

Closes #4290

### CODE REVIEW

**Comment by DaughterOfMars** [comment]
Run here: https://github.com/iotaledger/iota/actions/runs/12278874595/job/34261557947

Two tests are currently failing. I can address these issues here to wait for a subsequent PR.

**Comment by thibault-martinez** [comment]
Is the only difference with the PR simtest job that it runs them all?

**Comment by DaughterOfMars** [comment]
> Is the only difference with the PR simtest job that it runs them all?

`simtest_run.sh` does a lot of stuff with benchmarking and such that is not done by running the simtest command alone

File: .github/workflows/simulator_nightly.yml [hunk 1]
```diff
@@ -20,159 +20,42 @@ concurrency:
...
+  # from transient network timeouts or other issues.
+  CARGO_NET_RETRY: 10
+  RUSTUP_MAX_RETRIES: 10
+  # Don't emit giant backtraces in the CI logs.
+  RUST_BACKTRACE: short
+  RUST_LOG: off
+  NUM_CPUS: 24
 
 jobs:
   simtest:
```
Comment (thibault-martinez) [inline .github/workflows/simulator_nightly.yml]: Could it not be a job of `nightly.yml`?

**Review by thibault-martinez — APPROVED** [review]

**Comment by thibault-martinez** [comment]
> Run here: https://github.com/iotaledger/iota/actions/runs/12278874595/job/34261557947
> 
> Two tests are currently failing. I can address these issues here to wait for a subsequent PR.

Please open issues for the failing ones.

File: .github/workflows/simulator_nightly.yml (hunk 1 above)
Comment (DaughterOfMars) [reply .github/workflows/simulator_nightly.yml]: Sure, it totally could. Is that what we want to do?

**Comment by DaughterOfMars** [comment]
Issue for failing tests: https://github.com/iotaledger/iota/issues/4477

File: .github/workflows/simulator_nightly.yml (hunk 1 above)
Comment (thibault-martinez) [reply .github/workflows/simulator_nightly.yml]: I guess it makes sense now that it's extra small

Comment (DaughterOfMars) [reply .github/workflows/simulator_nightly.yml]: I combined them. I also stopped the reusable rust tests workflow from running its simtests as these should be covered by this one

**Review by thibault-martinez — APPROVED** [review]

**Review by miker83z — APPROVED** [review]

**Pull request merged**